lib/defender/aws_iot_defender.c

```diff
@@ -251,41 +268,56 @@ static void _metricsPublishTimerExpirationRoutine( void * pArgument )
 {
```

Comment:
This routing *could* (it won't, but could...) fire while the previous one is executing... you should atomically (if possible) set a flag that indicates that metrics collection and upload is in progress and therefore if it fires agaon it should do nothing and let the previous one terminate. Just so that the code is formally correct and an occasional reviewer that does not know we have a 5 minutes minimum interval will not be surprised.

Reply:
I agree totally. I will fix it.

Reply:
A second thought: let's deal with it after we have the plan for moving to task pool.